eclass/kernel-install.eclass

```diff
@@ -459,15 +632,29 @@ kernel-install_install_all() {
 	local dir_ver=${1}
 	local kernel_dir=${EROOT}/usr/src/linux-${dir_ver}
 	local relfile=${kernel_dir}/include/config/kernel.release
+	local image_path=$(dist-kernel_get_image_path)
 	local module_ver
 	module_ver=$(<"${relfile}") || die
 
+	if [[ ${KERNEL_IUSE_GENERIC_UKI} ]]; then
+		local uki="${image_path%/*}/uki.efi"
+		local initrd="${image_path%/*}/initrd"
+		if use generic-uki; then
+			$(tc-getOBJCOPY) -O binary -j.initrd "${kernel_dir}/${uki}" \
+				"${kernel_dir}/${initrd}" \
+					|| die "Failed to extract initramfs"
+		fi
+
+		if [[ -f ${kernel_dir}/${uki} ]]; then
+			kernel-install_extract_kernel "${kernel_dir}/${image_path}"
+		fi
+	fi
```

Comment:
Is there any reason not to put this all under `&& use generic-uki`? IIUC, with `! use generic-uki`, we won't have a UKI here.

Reply:
> Is there any reason not to put this all under `&& use generic-uki`? IIUC, with `! use generic-uki`, we won't have a UKI here.

I thought there was a reason, but I honestly can't remember so I'll try putting this in one block to see if this breaks anything

Reply:
Ooh now I remember, in `gentoo-kernel` (non-bin), this is the frist time this function gets called. We then want to extract the kernel image (regardless of USE=generic-uki or not).

Whereas in gentoo-kernel-bin, this is the second time this function is called, the second time will fail if the first call already removed the `uki.efi` (i.e. `USE=-generic-uki`).

Putting this in a conditional for the file existing makes it work in both cases.